### PR TITLE
feat: new project dialog for web

### LINKED ISSUE #446: New Project Dialog for Web
In web, the app uses the hard-coded "Project Web" project name instead of asking the user for a name.

This is a follow-up after #425

### PR DESCRIPTION
**Pull request by cguedes**
This closes #446 

This PR introduces a new core atom for modals named `createModalAtoms`.

With that, we can create a modal atoms for the create project model with:

```ts
export const createProjectModalAtoms = createModalAtoms<string>();
```

And then open the modal using:

```ts
const modalResult = useSetAtom(createProjectModalAtoms.openAtom);
if (modalResult.status === 'dismissed') {
   // modal closed or canceled
}
const projectName = modalResult.value;
...
```

That would return a string with the modal return value (a string), or `null` if the modal was closed/canceled.

The modal code would be something like:

```ts
  const isVisible = useAtomValue(createProjectModalAtoms.visibleAtom);
  const closeModal = useSetAtom(createProjectModalAtoms.closeAtom);
  const dismissModal = useSetAtom(createProjectModalAtoms.dismissAtom);
  ...
  const handleFormSubmit: FormEventHandler = (e) => {
    ...
    closeModal(name.trim());
  };
  const handleOnCancel: MouseEventHandler<HTMLButtonElement> = (e) => {
    dismissModal();
  };

```

image

image

### CODE REVIEW

File: src/application/CreateProjectModal.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    closeModal(null);
+  };
+
+  if (!isVisible) {
+    return <></>;
+  }
+
+  return (
+    <div className="fixed left-0 top-0 z-modals flex h-screen w-screen items-center justify-center">
+      <div className="pointer-events-none absolute left-0 top-0 h-full w-full bg-slate-400/60" />
```
Comment (sehyod) [inline src/application/CreateProjectModal.tsx:37]: A generic Modal component will be added in #479. Do you want to wait for it before merging this PR?

File: src/components/Button.tsx [hunk 2]
```diff
@@ -42,7 +46,8 @@ export function Button({
         className,
       )}
       disabled={disabled}
-      onClick={disabled ? undefined : onClick}
+      type={submit ? 'submit' : 'button'}
+      onClick={disabled || !onClick ? undefined : onClick}
```
Comment (sehyod) [inline src/components/Button.tsx:50]: I don't think `|| !onClick` is necessary here

File: src/application/CreateProjectModal.tsx (hunk 1 above)
Comment (cguedes) [reply src/application/CreateProjectModal.tsx:37]: I will.

File: src/components/Button.tsx (hunk 2 above)
Comment (cguedes) [reply src/components/Button.tsx:50]: Sure!

**Review by sehyod — APPROVED** [review]

File: src/application/CreateProjectModal.tsx [hunk 3]
```diff
@@ -1,65 +1,44 @@
...
-      <div className="pointer-events-none absolute left-0 top-0 h-full w-full bg-slate-400/60" />
-      <div
-        className={cx(
-          'relative w-[1150px] max-w-[calc(30vw-100px)] overflow-hidden',
-          'rounded-xl border-none border-slate-500 bg-white shadow-xl shadow-slate-500',
-          '',
-        )}
-      >
+    <Modal className="w-[1150px] max-w-[calc(30vw-100px)]" open={isVisible} onClose={() => dismissModal()}>
+      <div className="flex w-full flex-col  bg-white">
```
[the remark this answers is not in the document]
Comment (cguedes) [reply src/application/CreateProjectModal.tsx:32]: Great catch. Will fix that in the next issue for #447

**Pull request merged**